lib/samples/set_reference_scale/set_reference_scale_sample.dart

```diff
@@ -0,0 +1,227 @@
+  var _map = ArcGISMap();
+  // Create a variable to store the scale.
+  var _scale = 250000.0;
+  // Create a variable to store the ready state.
+  var _ready = false;
+  // Create a variable to store the selected feature layer source.
+  int? _selectedScale;
+
+  @override
+  void initState() {
+    super.initState();
+    // Add scales to the list.
+    _referenceScaleList.addAll([
+      DropdownMenuItem(
+        value: _possibleReferenceScales[0],
+        child: const Text('1:500,000'),
+      ),
+      DropdownMenuItem(
+        value: _possibleReferenceScales[1],
+        child: const Text('1:250,000'),
+      ),
+      DropdownMenuItem(
+        value: _possibleReferenceScales[2],
+        child: const Text('1:100,000'),
+      ),
+      DropdownMenuItem(
+        value: _possibleReferenceScales[3],
+        child: const Text('1:50,000'),
+      )
+    ]);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: SafeArea(
+        top: false,
+        // Add a column to the widget tree.
+        child: Column(
+          children: [
+            Expanded(
+              // Add a map view to the widget tree and set a controller.
+              child: ArcGISMapView(
+                controllerProvider: () => _mapViewController,
+                onMapViewReady: onMapViewReady,
+              ),
+            ),
+            // Add a Settings button to the widget tree.
+            ElevatedButton(
+              // Show the settings dialog when the button is pressed.
+              onPressed: _ready
+                  ? () => showDialog(
+                        context: context,
+                        builder: (context) => showSettings(context),
+                      )
+                  : null,
+              child: const Text(
+                'Settings',
+                style: TextStyle(
+                  color: Colors.deepPurple,
+                ),
+              ),
+            )
+          ],
+        ),
+      ),
+    );
+  }
+
+  void onMapViewReady() async {
+    // Create a portal item.
+    final portal = Portal.arcGISOnline();
+    final portalItem = PortalItem.withPortalAndItemId(
+        portal: portal, itemId: '3953413f3bd34e53a42bf70f2937a408');
+    // Load the portal item.
+    await portalItem.load();
+
+    // Create a map from the portal item and load it.
+    _map = ArcGISMap.withItem(portalItem);
+    await _map.load();
+
+    // Get the operational layers from the map.
+    for (var layer in _map.operationalLayers) {
+      _allFeatureLayers.add(layer.name);
+    }
+    _mapViewController.arcGISMap = _map;
+
+    // See if the map is ready.
+    setState(() => _ready = true);
+  }
+
+  StatefulBuilder showSettings(BuildContext newContext) {
+    // Show the settings dialog.
+    return StatefulBuilder(builder: (newContext, setNewState) {
+      return Dialog(
+        child: Padding(
+          padding: const EdgeInsets.all(8),
+          child: Column(
```

Comment:
What happens when you rotate the device into landscape? This won't all fit. I think it needs a SingleChildScrollView around the whole thing.